terraform/vpc/subnet.tf

```diff
@@ -0,0 +1,38 @@
+resource "aws_internet_gateway" "default" {
+  vpc_id = "${aws_vpc.myvpc.id}"
+}
+
+resource "aws_route_table" "internet" {
+  vpc_id = "${aws_vpc.myvpc.id}"
+  route {
+    cidr_block = "0.0.0.0/0"
+    gateway_id = "${aws_internet_gateway.default.id}"
+  }
+}
+
+resource "aws_route_table" "infra" {
+  vpc_id = "${aws_vpc.myvpc.id}"
+  route {
+    cidr_block = "0.0.0.0/0"
+    gateway_id = "${aws_internet_gateway.default.id}"
+  }
+}
+
+resource "aws_subnet" "infra" {
+  count             = "${var.infra_subnet_zone_count}"
+  vpc_id            = "${aws_vpc.myvpc.id}"
+  cidr_block        = "${lookup(var.infra_cidrs, concat("zone", count.index))}"
+  availability_zone = "${lookup(var.zones,       concat("zone", count.index))}"
+  map_public_ip_on_launch = true	# Really?
```

Comment:
I have actually removed this. Looks like it got back via the multiple rebases we did. I'll leave to decide this to whoever is using infra subnet first...